backend/hubspot/hubspot.go

```diff
@@ -500,28 +541,37 @@ func (h *HubspotApi) CreateCompanyForWorkspaceImpl(ctx context.Context, workspac
 			WithField("domain", domain).
 			Infof("company already exists in Hubspot. updating")
 
-		_, er := h.hubspotClient.Companies().Update(*companyID, companyProperties)
-		if er != nil {
-			return nil, er
+		if _, err := h.hubspotClient.Companies().Update(*companyID, companyProperties); err != nil {
+			return nil, err
 		}
+	} else {
+		log.WithContext(ctx).
+			WithField("name", name).
+			Warnf("failed to get client-side hubspot company. creating")
 
-		return
+		resp, err := h.hubspotClient.Companies().Create(companyProperties)
+		if err != nil {
+			return nil, err
+		}
+		companyID = &resp.CompanyID
+		log.WithContext(ctx).Infof("succesfully created a hubspot company with id: %v", resp.CompanyID)
 	}
 
-	log.WithContext(ctx).
-		WithField("name", name).
-		Warnf("failed to get client-side hubspot company. creating")
-
-	resp, err := h.hubspotClient.Companies().Create(companyProperties)
-	if err != nil {
-		return nil, err
+	if err = h.db.Model(&model.Workspace{Model: model.Model{ID: workspaceID}}).
+		Updates(&model.Workspace{HubspotCompanyID: companyID}).Error; err != nil {
+		return
 	}
-	log.WithContext(ctx).Infof("succesfully created a hubspot company with id: %v", resp.CompanyID)
-	if err := h.db.Model(&model.Workspace{Model: model.Model{ID: workspaceID}}).
-		Updates(&model.Workspace{HubspotCompanyID: &resp.CompanyID}).Error; err != nil {
-		return &resp.CompanyID, err
+
+	if adminEmail != "" {
+		admin := model.Admin{}
+		if err = h.db.Model(&model.Admin{}).Where(&model.Admin{Email: &adminEmail}).First(&admin).Error; err != nil {
+			return
+		}
+		if err := h.CreateContactCompanyAssociation(ctx, admin.ID, workspaceID); err != nil {
```

Comment:
requests the association to be created async so that the contact / company can be created first